web/src/lib/components/shared-components/search-bar/search-filter-box.svelte

```diff
@@ -208,18 +218,98 @@
+        takenBefore: undefined,
       },
-      inArchive: undefined,
-      inFavorite: undefined,
-      notInAlbum: undefined,
+      isArchive: undefined,
+      isFavorite: undefined,
+      isNotInAlbum: undefined,
       mediaType: MediaType.All,
     };
   };
 
-  const search = () => {};
+  const search = async () => {
+    let type: AssetTypeEnum | undefined = undefined;
+
+    if (filter.mediaType === MediaType.Image) {
+      type = AssetTypeEnum.Image;
+    } else if (filter.mediaType === MediaType.Video) {
+      type = AssetTypeEnum.Video;
+    }
+
+    let payload: SmartSearchDto | MetadataSearchDto = {
+      country: filter.location.country?.value,
+      state: filter.location.state?.value,
+      city: filter.location.city?.value,
+      make: filter.camera.make?.value,
+      model: filter.camera.model?.value,
+      takenAfter: filter.date.takenAfter
+        ? DateTime.fromFormat(filter.date.takenAfter, 'yyyy-MM-dd').toUTC().startOf('day').toString()
+        : undefined,
+      takenBefore: filter.date.takenBefore
+        ? DateTime.fromFormat(filter.date.takenBefore, 'yyyy-MM-dd').toUTC().endOf('day').toString()
+        : undefined,
+      isArchived: filter.isArchive ? filter.isArchive : undefined,
+      isFavorite: filter.isFavorite ? filter.isFavorite : undefined,
+      isNotInAlbum: filter.isNotInAlbum ? filter.isNotInAlbum : undefined,
+      personIds: filter.people && filter.people.length > 0 ? filter.people.map((p) => p.id) : undefined,
+      type,
+    };
+
+    if (filter.context) {
+      if (payload.personIds && payload.personIds.length > 0) {
+        handleError(
+          new Error('Context search does not support people filter'),
+          'Context search does not support people filter',
+        );
+        return;
+      }
+
+      payload = {
+        ...payload,
+        query: filter.context,
+      };
+    }
+
+    dispatch('search', payload);
+  };
+
+  function populateExistingFilters() {
+    if ($searchQuery) {
+      filter = {
+        context: 'query' in $searchQuery ? $searchQuery.query : '',
+        people:
+          'personIds' in $searchQuery ? ($searchQuery.personIds?.map((id) => ({ id })) as PersonResponseDto[]) : [],
+        location: {
+          country: $searchQuery.country ? { label: $searchQuery.country, value: $searchQuery.country } : undefined,
+          state: $searchQuery.state ? { label: $searchQuery.state, value: $searchQuery.state } : undefined,
+          city: $searchQuery.city ? { label: $searchQuery.city, value: $searchQuery.city } : undefined,
+        },
+        camera: {
+          make: $searchQuery.make ? { label: $searchQuery.make, value: $searchQuery.make } : undefined,
+          model: $searchQuery.model ? { label: $searchQuery.model, value: $searchQuery.model } : undefined,
+        },
+        date: {
+          takenAfter: $searchQuery.takenAfter
+            ? DateTime.fromISO($searchQuery.takenAfter).toUTC().toFormat('yyyy-MM-dd')
```

Comment:
I wonder if it is possible to somehow have the dates displayed in short formats in US and UK formats